src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -447,47 +497,64 @@ public Response get(String json) throws Exception
 		catch (JsonSyntaxException e)
 		{
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return builder.addError(e.getMessage()).build();
 		}
 		catch (QueryException e)
 		{
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return builder.addError(e.getMessage()).build();
 		}
 		catch (BeanValidationException e)
 		{
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return builder.addErrors(e.getErrorMessages()).build();
 		}
 		catch (MaxRowKeysForQueryExceededException e) {
 			logger.error("Query failed with too many rows", e);
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return builder.addError(e.getMessage()).build();
 		}
 		catch (MemoryMonitorException e)
 		{
 			logger.error("Query failed.", e);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			Thread.sleep(1000);
 			System.gc();
 			return setHeaders(Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(new ErrorResponse(e.getMessage()))).build();
 		}
 		catch (IOException e)
 		{
 			logger.error("Failed to open temp folder "+datastore.getCacheDir(), e);
+			Tags.ERROR.set(span, Boolean.TRUE);
+			span.log(e.getMessage());
 			return setHeaders(Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(new ErrorResponse(e.getMessage()))).build();
 		}
 		catch (Exception e)
 		{
 			logger.error("Query failed.", e);
+			Tags.ERROR.set(span, Boolean.TRUE);
```

Comment:
We need to add specific tag here signaling ``MaxRowKeys`` errors. Would be interesting to filter for those using that tag. We have another ``catch`` for this specific exception but it doesn't work, probably a different exception is raised.

Reply:
I have noticed that we are not consistent on how we set tags. Perhaps ``span.setTag`` is more consistent rather than ``Tags.ERROR.set`` style?

Reply:
We do see the MaxRowKeyException in the span logs - have at this one trace: https://app.lightstep.com/Production/trace?span_guid=7998ca7d2b4b3169&operation_name=datapoints_query&at_micros=1530615097721844#span-7998ca7d2b4b3169. Currently the number of row keys count in added as tag to this span : query_database_datapoints_count. Does this help?

Reply:
Please remove vendor specific info from the PR discussion.

Logs are not searchable and I feel we are mixing both here:

1. I want to be able to search for all spans which hit ``MaxRowKeys`` - or along these semantics.
2. It is not useful to search for high cardinality fields (*_count) because the values are not deterministic. (should I search for all possible count values from 1 till MAX_INT?)

the ``_count`` is useful, but it is sufficient to be available in the ``log``.

Reply:
Adding a span tag as
span.setTag("MaxRowKeys", Boolean.TRUE)
Regarding _count tags once we have better search feature than should we change the instrumentation again? For me the search works (though it's not in best shape currently) through filtering/autocomplete.